GeneratorInterface/Pythia8Interface/plugins/LHAupLesHouches.cc

```diff
@@ -93,11 +93,12 @@ bool LHAupLesHouches::setEvent(int inProcId)
                 hepeup.PUP[i][4], hepeup.VTIMUP[i],
                 hepeup.SPINUP[i],scalein);
   }
-  
+
   if(!infoPtr->eventAttributes) {
     fEvAttributes->clear();
     infoPtr->eventAttributes = fEvAttributes;
   } else {
+    infoPtr->eventAttributes = fEvAttributes;  // make sure still there
     infoPtr->eventAttributes->clear();
   }
```

Comment:
Just to note that now both branches do the same, so the if-else is redundant.

Reply:
Indeed they are redundant. Should I remove it?.
It is already merged

Reply:
@alberto-sanchez strictly speaking the code is not ideal and @makortel is right, but the result looks correct and still I think this is a workaround (although I need to look into pythia8 better). In case we want to clean it further you may just have a small PR, and then do the backport.

Reply:
Of course if it the real fix might need different code paths for the cases, a temporary cleanup probably doesn't make sense.

Reply:
actually as written the branches are different.. Perhaps the difference is a no-op, but then the code is rather confusing if that is the case.